osu.Game/Screens/Select/Carousel/CarouselGroupEagerSelect.cs

```diff
@@ -55,18 +55,25 @@ public override void RemoveChild(CarouselItem i)
                 updateSelectedIndex();
         }
 
+        private bool addingChildren;
+
         public void AddChildren(IEnumerable<CarouselItem> items)
         {
+            addingChildren = true;
+
             foreach (var i in items)
-                base.AddChild(i);
+                AddChild(i);
+
+            addingChildren = false;
 
             attemptSelection();
         }
 
         public override void AddChild(CarouselItem i)
         {
             base.AddChild(i);
-            attemptSelection();
+            if (!addingChildren)
+                attemptSelection();
```

Comment:
What's this guard about? Is it to protect against a potentially unsafe enumeration of `Children` inside `attemptSelection()`? If so then that check should probably be inside of that method because there are other usages of `attemptSelection()` that are potentially unsafe (filtering, child item state change).

Reply:
it's actually a performance concern if it's called too often